### ISSUE TITLE
Get the original T2I training dataset

### ISSUE THREAD

**Issue by iClaude1998** [OpenSenseNova/SenseNova-Vision#8]
---

Hi — we are reproducing the `t2i_general` training inputs of SenseNova-Vision and are stuck on one thing that
only you can answer. **We are not asking for images, latents, teacher trajectories or checkpoints.** We only
need to prove that a given caption belongs to the original training records, and to reproduce the text the
model was actually fed.

### What blocks us

`data/dataset_info.py` (L741-759, group `t2i_general`, loader `T2IJSONLIterableDataset`) declares three entries
whose `jsonl_path` is the empty string:

| entry | `data_dir` | `num_total_samples` | yaml `num_used_data` |
|---|---|---|---|
| `BLIP3o-Pretrain-Long-Caption` | `datas/train_data/blip3o/long_caption` | 6,847,620 | 2,000,000 |
| `BLIP3o-Pretrain-Short-Caption` | `datas/train_data/blip3o/short_caption` | 2,386,046 | 2,000,000 |
| `BLIP3o-Long-part2` | `datas/train_data/blip3o/long_caption` | 12,809,032 | 2,000,000 |

`data/configs/cv_unify/cv_unify_baseline_v9.yaml` L241-261 lists exactly these three names, and the repository
contains no converter that produces the three JSONLs, while `tools/data_prepare/` ships converters for the other
groups. `data/jsonl_dataset.py` L52-57 (`readlines()` → optional seed-0 shuffle → `[:num_used_data]`) therefore
cannot be replayed: without the original line order, "these 2,000,000 rows were the trained ones" is unprovable.

### What we checked ourselves first

1. **The `sa_*`-family partition hypothesis is refuted by measurement.** `sa_000000.tar` holds **11,183** rows,
   not ~6,848; `1000 × 11,183 = 11,183,000 ≠ 6,847,620`, and `6,847,620 / 11,183 = 612.3` is not an integer.
   The first ten `sa_*` tars average ≥ 10,190 rows. So `BLIP3o-Pretrain-Long-Caption` is not the `sa_*` family
   and `BLIP3o-Long-part2` is not "the rest".
2. **The HF datasets-server cannot decide it.** Its conversion is partial (101,900 Long rows = 10 `sa_*` tars;
   29,100 Short rows), so offsets at and around 6,847,620 return "list index out of range" and no hypothesis
   touching `webdataset_JDB_*` / `webdataset_shard_*` rows is testable.
3. **The counts imply an undocumented filter.** Short's 2,386,046 is ≈49% of the public pool
   (1,831 tars × ~2,646 ≈ 4.84M); Long's 6,847,620 + 12,809,032 = 19,656,652 is below the ~27M the public
   README claims.
4. `tools/data_prepare/README.md` L48 promises that "unresolved subset differences" are documented;
   `docs/train_data_prepare.md` at this commit does not contain that statement.

### What would close it (any one of 1-5, or the fallback 6)

1. The ordered manifest, or a stable row-id list, actually used for each `t2i_general` source.
2. The exact definition of `BLIP3o-Long-part2` vs `BLIP3o-Pretrain-Long-Caption` — shards / index ranges /
   filter, whether disjoint, with counts reproducing **6,847,620** and **12,809,032**.
3. The shard list, row ranges or member ids actually trained for Long-Caption and Short-Caption.
4. The caption field rules: which field, and the cleaning / concatenation / normalization applied before it
   became `conversations[0]["value"]`. (`data/t2i_dataset_jsonl.py` L50-59 reads it verbatim and applies no
   template of its own, so any normalization happened upstream of the JSONL.)
5. A stable mapping from each trained row to a public BLIP3o row — tar name + `__key__` is enough — plus the
   `image` relative-path convention the JSONL used.
6. **If the manifests cannot be released:** a hash of the member list (e.g. sha256 over the sorted
   `tar_name/__key__` lines) per entry, **plus** per-shard row counts for every tar used, **plus** a
   deterministic reconstruction script we can run against the public repos and verify against those hashes.

### References

- https://github.com/OpenSenseNova/SenseNova-Vision/blob/884d5151c101dda8e46ff728756f1c74e796c35c/docs/train_data_prepare.md
- https://github.com/OpenSenseNova/SenseNova-Vision/blob/884d5151c101dda8e46ff728756f1c74e796c35c/data/dataset_info.py
- https://huggingface.co/datasets/BLIP3o/BLIP3o-Pretrain-Long-Caption (rev `e4d07091a466d1a1e35a9b0c61caddc78d14a059`, 2,891 tars)
- https://huggingface.co/datasets/BLIP3o/BLIP3o-Pretrain-Short-Caption (1,831 tars)

Thanks very much — happy to do the reconstruction work ourselves; we just need the rule.

**Comment by Robin970822** [comment]
Thank you for the detailed investigation and for clearly documenting the reproducibility gap.

We have now published the materials needed to reconstruct the three `t2i_general` JSONLs:

- [SenseNova-Vision-Corpus-50M release](https://huggingface.co/datasets/sensenova/SenseNova-Vision-Corpus-50M)
- [GitHub update](https://github.com/OpenSenseNova/SenseNova-Vision/commit/743c59fe237c050b4998705d89683a068e545862)
- [BLIP3o reconstruction guide](https://github.com/OpenSenseNova/SenseNova-Vision/blob/master/tools/data_prepare/image_generation/README.md)
- [`convert_blip3o_keys.py`](https://github.com/OpenSenseNova/SenseNova-Vision/blob/master/tools/data_prepare/image_generation/convert_blip3o_keys.py)
- [End-to-end training data preparation guide](https://github.com/OpenSenseNova/SenseNova-Vision/blob/master/docs/train_data_prepare.md#general-understanding-generation-and-editing)

### Released key lists

| Training entry | Records | Released key list | Upstream namespace |
| --- | ---: | --- | --- |
| `BLIP3o-Pretrain-Short-Caption` | 2,386,046 | [Short Caption keys](https://huggingface.co/datasets/sensenova/SenseNova-Vision-Corpus-50M/blob/main/image_generation/keys/BLIP3o-Pretrain-Short-Caption.keys.txt.gz) | `BLIP3o/BLIP3o-Pretrain-Short-Caption` |
| `BLIP3o-Pretrain-Long-Caption` | 6,847,620 | [Long Caption keys](https://huggingface.co/datasets/sensenova/SenseNova-Vision-Corpus-50M/blob/main/image_generation/keys/BLIP3o-Pretrain-Long-Caption.keys.txt.gz) | `BLIP3o/BLIP3o-Pretrain-Long-Caption` |
| `BLIP3o-Pretrain-Long-Caption-part2` | 12,809,032 | [Long Caption part2 keys](https://huggingface.co/datasets/sensenova/SenseNova-Vision-Corpus-50M/blob/main/image_generation/keys/BLIP3o-Pretrain-Long-Caption-part2.keys.txt.gz) | `BLIP3o/BLIP3o-Pretrain-Long-Caption` |

Long Caption and Long Caption part2 are separate selections from the same public BLIP3o Long Caption source, so both use the `BLIP3o-Pretrain-Long-Caption` canonical namespace.

### Caption provenance

All three JSONLs were constructed entirely from the original captions distributed in the public BLIP3o archives.

For every selected image key, the caption comes from its paired `.txt` member in the corresponding BLIP3o tar archive. We did not apply caption rewriting, concatenation, or model-generated replacement. The released converter decodes the original text as UTF-8 and removes only surrounding whitespace before writing it to `conversations[0]["value"]`.

### Ordering and exact reproduction

The original JSONLs were constructed concurrently. Therefore, their row order and image-relative-path order do **not** strictly follow:

- the Hugging Face repository file order;
- lexicographic tar/shard order; or
- member order inside each tar archive.

Consequently, the Long Caption and Long Caption part2 selections should not be interpreted as contiguous ranges or a simple split of the ordered BLIP3o tar stream.

The construction order should have limited effect on training because the records were globally shuffled by the training data loader. Nevertheless, exact pre-shuffle ordering is necessary to reproduce the same post-shuffle `num_used_data` selection.

For that reason, each released key list preserves the exact order of the corresponding JSONL used for training **before the training-time global shuffle**. The converter preserves this key-list order when rebuilding the JSONL. Each canonical key also identifies the upstream namespace, tar stem, and image member, for example:

```text
BLIP3o/BLIP3o-Pretrain-Short-Caption/00000/000000524.jpg
BLIP3o/BLIP3o-Pretrain-Long-Caption/sa_000912/sa_10204894.jpg
